### PR TITLE
update for gradle 7.x compatibility

### PR DESCRIPTION
**Pull request by johnpooley**
This is to (hopefully) fix the current lack of compatibility with Gradle 7.x

### CODE REVIEW

**Comment by robert-mirzakhanian** [comment]
@lreimer Hi!
Please check this changes. =)

File: src/test/groovy/de/qaware/gradle/plugin/xsd2java/xjc/Xsd2JavaPluginTest.groovy [hunk 1]
```diff
@@ -19,8 +19,8 @@ import de.qaware.gradle.plugin.xsd2java.xjc.tasks.XSD2JavaTask
 import org.gradle.api.Task
 import org.gradle.testfixtures.ProjectBuilder
 import org.gradle.testkit.runner.GradleRunner
-import org.junit.Rule
-import org.junit.rules.TemporaryFolder
+import org.gradle.internal.impldep.org.junit.Rule
```
Comment (lreimer) [inline src/test/groovy/de/qaware/gradle/plugin/xsd2java/xjc/Xsd2JavaPluginTest.groovy]: Don' think that depending on a Gradle internal API is such a good idea. Is there another option?

Comment (robert-mirzakhanian) [reply src/test/groovy/de/qaware/gradle/plugin/xsd2java/xjc/Xsd2JavaPluginTest.groovy]: Hi @johnpooley! In org.spockframework:spock-core:2.0-groovy-3.0 deleted junit. What do you think about add dependency testImplementation 'junit:junit:4.13.2' and revert changes into Xsd2JavaPluginTest?

Comment (lreimer) [reply src/test/groovy/de/qaware/gradle/plugin/xsd2java/xjc/Xsd2JavaPluginTest.groovy]: Already working on the PR. 
The solution is to use the following dependency: `testImplementation 'org.spockframework:spock-junit4:2.0-groovy-3.0'`

Comment (johnpooley) [reply src/test/groovy/de/qaware/gradle/plugin/xsd2java/xjc/Xsd2JavaPluginTest.groovy]: hello, just reading this now. Added this change to the PR here too

**Comment by lreimer** [comment]
Merge the PR and made a few required additional changes.

**Comment by robert-mirzakhanian** [comment]
Hi @lreimer!
It is great news about new version!
Can you update information in README.md and add new version in https://plugins.gradle.org

**Pull request merged**